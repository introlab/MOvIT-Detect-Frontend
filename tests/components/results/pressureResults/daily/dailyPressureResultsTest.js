/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import DailyPressureResults from '../../../../../src/components/results/pressureResults/daily/dailyPressureResults';

Enzyme.configure({ adapter: new Adapter() });


describe('DailyPressureResults Tests', () => {
  const date = 1517720400000;
  let wrapper;

  const initialState = {
    applicationReducer: { language: 'FR' },
    recommendationReducer: {
      reduceWeight: true,
      reduceSlidingRest: true,
      reduceSlidingMoving: true,
    },
  };
  const mockStore = configureMockStore();
  const store = mockStore(initialState);
  const props = {
    date,
  };

  beforeEach(() => {
    wrapper = shallow(<DailyPressureResults store={store} {...props} />).dive();
  });

  it('should have proptypes', () => {
    const actualValue = DailyPressureResults.propTypes;

    const expectedValue = {
      language: PropTypes.string.isRequired,
      date: PropTypes.instanceOf(Date),
      reduceWeight: PropTypes.bool,
      reduceSlidingMoving: PropTypes.bool,
      reduceSlidingRest: PropTypes.bool,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
