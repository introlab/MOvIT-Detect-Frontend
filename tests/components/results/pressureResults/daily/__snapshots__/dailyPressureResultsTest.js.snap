// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyPressureResults Tests should match the snapshot 1`] = `
<div>
  <div
    className="col-lg-2 leftMenu"
  >
    <ul
      className="graphlist"
    >
      <li
        className="graphLink"
      >
        <a
          href="results/pressure#dailyPressureCenter"
        >
          results.graphicsLink.pressureCenter.FR
        </a>
      </li>
      <li
        className="graphLink"
      >
        <a
          href="results/pressure#reduceWeight"
        >
          dailyResults.pressure.FR
        </a>
      </li>
    </ul>
  </div>
  <div
    className=" col-lg-10 offset-lg-3 results resultsContainer"
  >
    <div
      className="col-lg-8 graphic"
    >
      <div>
        <div>
          <Connect(PressureCenter)
            date={1517720400000}
            title="results.graphicsLink.pressureCenter.FR"
          />
          <div
            id="reduceWeight"
          >
            <Connect(RecGoalProgress)
              condition={true}
              goalValue={70}
              recValue={50}
              title="dailyResults.pressure.FR"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
